src/compiler/crystal/exception.cr

```diff
@@ -94,4 +99,194 @@ module Crystal
       end
     end
   end
+
+  module ErrorFormat
+    MACRO_LINES_TO_SHOW               = 3
+    OFFSET_FROM_LINE_NUMBER_DECORATOR = 6
+
+    def error_body(source, default_message) : String | Nil
+      case filename = @filename
+      when VirtualFile
+        return format_macro_error(filename)
+      when String
+        if File.file?(filename)
+          return format_error_from_file(filename)
+        end
+      end
+
+      return format_error(source) if source
+      default_message
+    end
+
+    def line_number_decorator(line_number)
+      " #{line_number} | "
+    end
+
+    def append_error_indicator(io, offset, column_number, size = 0)
+      size ||= 0
+      io << '\n'
+      io << (" " * (offset + column_number - 1))
+      with_color.green.bold.surround(io) do
+        io << '^'
+        if size > 0
+          io << ("-" * (size - 1))
+        end
+      end
+    end
+
+    def filename_row_col_message(filename, line_number, column_number)
+      String.build do |io|
+        io << colorize("#{relative_filename(filename)}:#{line_number}:#{column_number}").underline
+      end
+    end
+
+    def format_error(filename, lines, line_number, column_number, size = 0)
+      return "#{relative_filename(filename)}" unless line_number
+
+      unless line = lines[line_number - 1]?
+        return "in #{filename_row_col_message(filename, line_number, column_number)}"
+      end
+
+      String.build do |io|
+        io << "in "
+        case filename
+        when String
+          io << filename_row_col_message(filename, line_number, column_number)
+        when VirtualFile
+          io << "macro '" << colorize("#{filename.macro.name}").underline << '\''
+        else
+          io << "unknown location"
+        end
+
+        decorator = line_number_decorator(line_number)
+        lstripped_line = line.lstrip
+        space_delta = line.chars.size - lstripped_line.chars.size
+
+        io << "\n\n"
+        io << colorize(decorator).dim << colorize(lstripped_line.chomp).bold
+        append_error_indicator(io, decorator.chars.size, column_number - space_delta, size || 0)
+      end
+    end
+
+    def format_error_from_file(filename : String)
+      lines = File.read_lines(filename)
+      format_error(
+        filename: @filename,
+        lines: lines,
+        line_number: @line_number,
+        column_number: @column_number,
+        size: @size
+      ).capitalize
```

Comment:
This seems to be the problem of the downcase...

Reply:
Nice catch! Fixed.